src/rpc/server.cpp

```diff
@@ -18,6 +18,7 @@
 #include <memory> // for unique_ptr
 #include <unordered_map>
 
+const std::set<std::string>& GetWhitelistedRpcs(const std::string&);
```

Comment:
Didn't include `src/httprpc.h` because would add a circular dependency?

Reply:
Yes, because travis failed on two builds with such an error.

Reply:
I think it's probably better to add the #include and update test/lint/lint-circular-dependencies.sh.

Reply:
Was this resolved by mistake?

Reply:
Too many JavaScript-based buttons. ;)

Reply:
I don't think I should simply change the linter-checks because of a single function. Maybe someone more qualified should be asked for advice.

Reply:
Even nicer would be to solve the circular dependency in the first place. Using a forward declare here instead of an include isn't a solution - it's just hiding it from the linter.

If it's too much work to resolve that in this PR, my suggestion is being explicit about it and adding an exception to the linter.

Reply:
I'll try to extract the circular structures into another include and then recombine everything. Maybe better in a separate PR?

Reply:
So, I have extracted the following classes and functions into a new include **rpc/interfaces.h**

```cpp
class RPCTimerBase
class RPCTimerInterface
class CRPCCommand
class CRPCTable
std::string JSONRPCExecBatch(const JSONRPCRequest& jreq, const UniValue& vReq);
void RPCSetTimerInterface(RPCTimerInterface *iface);
void RPCSetTimerInterfaceIfUnset(RPCTimerInterface *iface);
void RPCUnsetTimerInterface(RPCTimerInterface *iface);
```

A test run with `circular-dependencies.py`script shows no httprpc->server->httprpc anymore (but there are other _unrelated_ circular dependencies still)

I have thought about extracting their implementations in **rpc/server.cpp** as well, but as this would look even more invasive, I'll remain with this single include file for now. 

I think, I should open a separate PR and let others discuss about it. And if it gets merged, I'll rebase this PR afterwards.

Reply:
I have *outsourced* this task into this PR (draft): https://github.com/bitcoin/bitcoin/pull/18871

Reply:
Will resolve this conversation if you don't mind. In case my other PR gets merged we can continue from there (with rebase etc.).